Xamarin.Forms.Platform.Android/Renderers/AndroidGIFImageParser.cs

```diff
@@ -264,6 +257,56 @@ public static async Task<IFormsAnimationDrawable> LoadImageAnimationAsync(UriIma
    
 			return animation;
 		}
+
+		internal static async Task<FormsAnimationDrawable> GetFormsAnimationDrawableFromResource(string resource, Context context, BitmapFactory.Options options)
+		{
+			FormsAnimationDrawable animation = null;
+
+			using (var stream = context.Resources.OpenRawResource(ResourceManager.GetDrawableByName(resource)))
+			using (var decoder = new AndroidGIFImageParser(context, options.InDensity, options.InTargetDensity))
+			{
+				try
+				{
+					if (!FileImageSourceHandler.DecodeSynchronously)
+						await decoder.ParseAsync(stream).ConfigureAwait(false);
+					else
+						decoder.ParseAsync(stream).Wait();
+
+					animation = decoder.Animation;
+				}
+				catch (GIFDecoderFormatException)
+				{
+					animation = null;
+				}
+			}
+
+			return animation;
+		}
+
+		internal static async Task<FormsAnimationDrawable> GetFormsAnimationDrawableFromFile(string file, Context context, BitmapFactory.Options options)
+		{
+			FormsAnimationDrawable animation = null;
+
+			using (var stream = new FileStream(file, FileMode.Open, FileAccess.Read, FileShare.Read, DefaultBufferSize, true))
```

Comment:
The code here looks identical to the code in the method above except that the stream is created differently.
Maybe add an other method which takes a stream instead of a string and call it from these two methods (`GetFormsAnimationDrawableFromResource`, `GetFormsAnimationDrawableFromFile`) to reduce code duplication.

Also it looks like the `GetFormsAnimationDrawableFromResource` method can directly take the `drawableIdentifier` so `ResourceManager.GetDrawableByName` does need to be called twice in this case.

Reply:
Right. I have created a new method that receives a Stream and is used in the other cases, when using a Resource or a file path.